src/rules/no-anonymous-default-export.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @fileoverview Rule to disallow anonymous default exports.
+ * @author Duncan Beevers
+ */
+
+const MAP = {
+  Literal: 'allow-literal',
+  ObjectExpression: 'allow-object-expression',
+  FunctionDeclaration: 'allow-anonymous-function-declaration',
+  ArrowFunctionExpression: 'allow-arrow-function-expression',
+  ClassDeclaration: 'allow-anonymous-class-declaration',
+}
+
+const FLAGS = []
+for (const nodeType in MAP) {
+  FLAGS.push(MAP[nodeType])
+}
+
+module.exports = {
+  meta: {
+    schema: {
+      type: 'array',
```

Comment:
I would be fine with simplifying the name of the properties (like ArrowFunctionExpression --> ArrowFunction), you'd just need to account for the difference for that in the code.